### PR TITLE
Angular factories for managing articles, pages, and topics

### PR DESCRIPTION
**Pull request by blakecallens**
Moves these pages to an async model

### CODE REVIEW

File: plugins/pencilblue/controllers/actions/admin/content/topics/import_topics.js [hunk 1]
```diff
@@ -91,7 +91,7 @@ module.exports = function(pb) {
                 });
             }
 
-            cb({content: pb.BaseController.apiResponse(pb.BaseController.API_SUCCESS, loc.topics.TOPICS_CREATED)});
+            cb({content: pb.BaseController.apiResponse(pb.BaseController.API_SUCCESS, self.ls.get('TOPICS_CREATED'))});
```
Comment (brianhyder) [inline plugins/pencilblue/controllers/actions/admin/content/topics/import_topics.js]: Please update this to ```self.ls.g('TOPICS_CREATED')```

File: plugins/pencilblue/controllers/admin/content/articles/manage_articles.js [hunk 2]
```diff
@@ -32,81 +32,20 @@ module.exports = function(pb) {
...
-            
-            var angularObjects = pb.ClientJs.getAngularObjects(
-                {
-                    navigation: pb.AdminNavigation.get(self.session, ['content', 'articles'], self.ls, self.site),
-                    pills: pills,
-                    articles: articles
-                });
-            //TODO: err first arg for style. User experience error when no pills?
-            cb(angularObjects);
+        self.setPageName(self.ls.get('MANAGE_ARTICLES'));
```
Comment (brianhyder) [inline plugins/pencilblue/controllers/admin/content/articles/manage_articles.js]: ```self.ls.g```

File: plugins/pencilblue/controllers/admin/content/pages/manage_pages.js [hunk 3]
```diff
@@ -33,68 +33,16 @@ module.exports = function(pb) {
...
-            
-            var angularObjects = pb.ClientJs.getAngularObjects({
-                navigation: pb.AdminNavigation.get(self.session, ['content', 'pages'], self.ls, self.site),
-                pills: pills,
-                pages: self.getPageStatuses(pagesWithAuthor)
-            });
-            
-            //TODO: err first arg for style. User experience error when no pills?
-            cb(angularObjects);
+        self.setPageName(self.ls.get('MANAGE_PAGES'));
```
Comment (brianhyder) [inline plugins/pencilblue/controllers/admin/content/pages/manage_pages.js]:
```
self.ls.g
```

File: plugins/pencilblue/controllers/admin/content/topics/manage_topics.js [hunk 4]
```diff
@@ -16,75 +16,31 @@
...
+            pills: self.getAdminPills(SUB_NAV_KEY, self.ls, SUB_NAV_KEY)
+        });
 
-            self.setPageName(self.ls.get('MANAGE_TOPICS'));
-            self.ts.registerLocal('angular_objects', new pb.TemplateValue(angularObjects, false));
-            self.ts.load('admin/content/topics/manage_topics', function(err, data) {
-                var result = '' + data;
-                cb({content: result});
-            });
+        self.setPageName(self.ls.get('MANAGE_TOPICS'));
```
Comment (brianhyder) [inline plugins/pencilblue/controllers/admin/content/topics/manage_topics.js]:
```
self.ls.g
```

File: plugins/pencilblue/controllers/api/content/article_api_controller.js [hunk 5]
```diff
@@ -41,26 +42,52 @@ module.exports = function(pb) {
...
+        this.service.getAllWithCount(options, function(err, obj) {
+            if (util.isError(err)) {
+                return cb(err);
+            }
+            else if (util.isNullOrUndefined(obj)) {
+                return self.notFound(cb);
+            }
+
+            var userService = new UserService(self.getServiceContext());
+            userService.getAuthors(obj.data, function(err, articlesWithAuthorNames) {
```
Comment (brianhyder) [inline plugins/pencilblue/controllers/api/content/article_api_controller.js]: This function along with the whole prototype ```UserService``` has been deprecated and replaced by ```UserServiceV2```.  Is this so we get the author information on the article list?

File: plugins/pencilblue/controllers/api/content/page_api_controller.js [hunk 6]
```diff
@@ -40,19 +41,45 @@ module.exports = function(pb) {
...
+        this.service.getAllWithCount(options, function(err, obj) {
+            if (util.isError(err)) {
+                return cb(err);
+            }
+            else if (util.isNullOrUndefined(obj)) {
+                return self.notFound(cb);
+            }
+
+            var userService = new UserService(self.getServiceContext());
+            userService.getAuthors(obj.data, function(err, pagesWithAuthorNames) {
```
Comment (brianhyder) [inline plugins/pencilblue/controllers/api/content/page_api_controller.js]: Same as the article_api_controller

File: plugins/pencilblue/templates/angular/admin/content/pages/manage_pages.html [hunk 7]
```diff
@@ -34,41 +42,116 @@
...
+    				else if(pages[i].publish_date > now) {
+    					pages[i].status = loc.articles.UNPUBLISHED;
+    				}
+    				else {
+    					pages[i].status = loc.articles.PUBLISHED;
+    				}
+    			}
+
+    			return pages;
+    		}
```
Comment (brianhyder) [inline plugins/pencilblue/templates/angular/admin/content/pages/manage_pages.html]: The code to handle articles and pages is really similar.  How hard would it be to condense this down so we don't have duplicate code?

File: plugins/pencilblue/controllers/api/content/article_api_controller.js (hunk 5 above)
Comment (blakecallens) [inline plugins/pencilblue/controllers/api/content/article_api_controller.js]: Yup.

Comment (blakecallens) [inline plugins/pencilblue/controllers/api/content/article_api_controller.js]: Should I use the getFullName function instead?

File: plugins/pencilblue/templates/angular/admin/content/pages/manage_pages.html (hunk 7 above)
Comment (blakecallens) [inline plugins/pencilblue/templates/angular/admin/content/pages/manage_pages.html]: I can add a service.

File: plugins/pencilblue/controllers/api/content/article_api_controller.js (hunk 5 above)
Comment (brianhyder) [inline plugins/pencilblue/controllers/api/content/article_api_controller.js]: There is an option that can be passed to the ```ArticleServiceV2.getAll``` function that will trigger this to happen.  I _think_ all you will have to do in the article api controller is:

```
ArticleApiController.prototype.processQuery = function() {
  var options = ArticleApiController.super_.prototype.processQuery.apply(this);
  options.render = !!this.query.render; //pass 1 for true, 0 or nothing for false
  return options;
};
```

Comment (brianhyder) [inline plugins/pencilblue/controllers/api/content/article_api_controller.js]: The render portion should allow you to get the full name piece.

File: plugins/pencilblue/templates/angular/admin/content/pages/manage_pages.html (hunk 7 above)
Comment (brianhyder) [inline plugins/pencilblue/templates/angular/admin/content/pages/manage_pages.html]: :+1:

File: plugins/pencilblue/controllers/api/content/article_api_controller.js (hunk 5 above)
Comment (blakecallens) [inline plugins/pencilblue/controllers/api/content/article_api_controller.js]: This worked for articles, but not pages.

File: plugins/pencilblue/controllers/actions/admin/content/topics/import_topics.js (hunk 1 above)
Comment (blakecallens) [inline plugins/pencilblue/controllers/actions/admin/content/topics/import_topics.js]: self.ls.g isn't working

**Pull request merged**